SCClassLibrary/Common/UnitTesting/UnitTest.sc

```diff
@@ -239,24 +242,30 @@ UnitTest {
 		server.newAllocators; // new nodes, busses regardless
 	}
 
+	debug { |text|
+		debug = debug ++ text;
+	}
+
 	// call failure directly
-	failed { | method, message, report = true |
-		var r = UnitTestResult(this, method, message);
+	failed { | method, message, report = true, details |
+		var r;
+		r = UnitTestResult(this, method, message, details, debug);
 		failures = failures.add(r);
 		if(report){
-			Post << Char.nl << "FAIL:";
+			Post << Char.nl << "FAIL: ";
 			r.report;
 			Post << Char.nl;
 		};
 	}
 
 	// call pass directly
-	passed { | method, message, report = true |
-		var r = UnitTestResult(this, method, message);
+	passed { | method, message, report = true, details |
+		var r;
+		r = UnitTestResult(this, method, message, details, debug);
 		passes = passes.add(r);
-		if(report and: reportPasses) {
-			Post << "PASS:";
-			r.report;
+		if(report and: reportPasses != false) {
```

Comment:
It would be better to use only symbols as selectors. Mixing booleans and symbols isn't very idiomatic and may be confusing. Also it may lead to errors in use when you expect a boolean and use it in an if-clause.

One standard solution is the introduction of verbosity levels.
`verbosity = 0` (`verbosity = 1`, `verbosity = 2` etc.)
or 
`verbosity = \detailed` (`verbosity = \brief`, `verbosity = \silent` etc.)

Reply:
Also, I would suggest bracketing here: "normal" languages interpret that as `report and: (reportPasses != false)` but SC's left-to-right precedence is `(report and: reportPasses) != false`. If you want the former, it must be written explicitly; if the latter, it's not strictly necessary but the parens make it clear that it's not a mistake.

For the former, short-circuiting notation is even better: `report and: { reportPasses != false }` (don't evaluate the second condition if `report` is already false).

Reply:
> It would be better to use only symbols as selectors. Mixing booleans and symbols isn't very idiomatic and may be confusing. Also it may lead to errors in use when you expect a boolean and use it in an if-clause.

I totally agree, but unfortunately this would be an API change, and I was trying hard to keep it backwards-compatible since [last time I attempted even a tiny API change in `UnitTest` I got called out on it](https://github.com/supercollider/supercollider/pull/3354#discussion_r158592264) ;-)  Any other suggestions?

> Also, I would suggest bracketing here: [snipped]

Whoops, good catch thanks!  I have to say, SC's left-to-right precedence does drive me crazy.  I *really* wish that could be fixed, because it's so insanely counter-intuitive for anyone who's programmed in pretty much any other language.  Sorry for the mini-rant ;-)

Reply:
> Any other suggestions?

Add an instance variable called `verbosity`. It is backward compatible.

do you want reasons for why left-to-right is good?